website/docs/plugin/terraform-plugin-protocol.mdx

```diff
@@ -0,0 +1,120 @@
+| `GetSchema`                  | `GetProviderSchema`          | Returns provider schema, provider metaschema, all resource schemas and all data source schemas.                                                                                                                                      |
+| `PrepareProviderConfig`      | `ValidateProviderConfig`     | Validates the practitioner supplied provider configuration by verifying types conform to the schema and allows for custom or advanced validation.                                                                                    |
+| `ValidateResourceTypeConfig` | `ValidateResourceConfig`     | Validates the practitioner supplied resource configuration by verifying types conform to the schema and allows for custom or advanced validation.                                                                                    |
+| `ValidateDataSourceConfig`   | `ValidateDataResourceConfig` | Validates the practitioner supplied data source configuration by verifying types conform to the schema and allows for custom or advanced validation.                                                                                 |
+| `UpgradeResourceState`       | `UpgradeResourceState`       | Called when a resource in state has a schema that doesn't match the current version. The provider is expected to modify the state to upgrade it to the latest schema.                                                                |
+| `Configure`                  | `ConfigureProvider`          | Passes the practitioner-specified provider configuration to the provider.                                                                                                                                                            |
+| `ReadResource`               | `ReadResource`               | Called when refreshing a resource's state.                                                                                                                                                                                           |
+| `ReadDataSource`             | `ReadDataSource`             | Called when refreshing a data source's state.                                                                                                                                                                                        |
+| `PlanResourceChange`         | `PlanResourceChange`         | Calculates a plan for a resource. A proposed new state is generated, which the provider can modify.                                                                                                                                  |
+| `ApplyResourceChange`        | `ApplyResourceChange`        | Called when a diff between the resource's state and configuration has been detected, and the practitioner has approved a planned change. The provider is to apply the changes contained in the plan, and return the resulting state. |
```

Comment:
Terraform allows for plans without configuration changes (e.g. only updating `Computed`-only values). This is so providers can signal potential provider behavior changes via a plan-apply workflow. I believe the below wording should capture what we are going for here.

```suggestion
| `ApplyResourceChange`        | `ApplyResourceChange`        | Called when a practitioner has approved a planned change. The provider is to apply the changes contained in the plan, and return a resulting state matching the given plan. |
```